File: molecule/default/tests/test_default.rb
# frozen_string_literal: true

describe command('/opt/node/bin/node --version') do
  its('stdout') { should eq "v8.11.3\n" }
end

describe command('node --version') do
  its('stdout') { should eq "v8.11.3\n" }
end

describe command('/opt/node/bin/npm --version') do
  its('stdout') { should eq "5.6.0\n" }
end

describe command('npm --version') do
  its('stdout') { should eq "5.6.0\n" }
end

describe command('/opt/node/bin/npx --version') do
  its('stdout') { should eq "9.7.1\n" }
end

describe command('npx --version') do
  its('stdout') { should eq "9.7.1\n" }
end

File: molecule/default/tests/test_default.py
import os

import testinfra.utils.ansible_runner

testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
    os.environ['MOLECULE_INVENTORY_FILE']
).get_hosts('all')


def test_nodejs(host):
    cmd = host.run("/opt/node/bin/node --version")
    assert cmd.stdout == """v8.11.3"""

    cmd = host.run("node --version")
    assert cmd.stdout == """v8.11.3"""

    cmd = host.run("/opt/node/bin/npm --version")
    assert cmd.stdout == """5.6.0"""

    cmd = host.run("npm --version")
    assert cmd.stdout == """5.6.0"""

    cmd = host.run("/opt/node/bin/npx --version")
    assert cmd.stdout == """9.7.1"""

    cmd = host.run("npx --version")
    assert cmd.stdout == """9.7.1"""

    assert host.file("/root/.npmrc").exists
